src/main/php/peer/http/Channel.class.php

```diff
@@ -0,0 +1,83 @@
+<?php namespace peer\http;
+
+use peer\SocketInputStream;
+
+class Channel implements \io\streams\InputStream {
+  private $socket;
+  private $reuseable= null;
+
+  /** @param peer.Socket */
+  public function __construct($socket) {
+    $this->socket= $socket;
+  }
+
+  /** @return peer.Socket */
+  public function socket() { return $this->socket; }
+
+  /**
+   * Rebinds to a new socket, closing the existing one if necessary
+   *
+   * @param  peer.Socket
+   * @return void
+   */
+  public function bind($socket) {
+    if ($this->socket->isConnected()) {
+      $this->socket->close();
+    }
+    $this->socket= $socket;
+  }
+
+  /**
+   * Connect (if necessary)
+   *
+   * @param  float $connectTimeout
+   * @param  float $readTimeout
+   * @return void
+   */
+  public function connect($connectTimeout, $readTimeout) {
+    if (false === $this->reuseable) {
+      $this->socket->close();
+    } else if ($this->socket->isConnected()) {
```

Comment:
Is this check enough? If the server closes the connection (e.g. because of a timeout on a reused connection), we would probably receive an `EOF` here, but `isConnected()` would still be true,

Reply:
Yes, this actually *is* a problem; and it seems `eof()` could help here. In need of some more real-life testing!